File: playlist-manager-ui/src/api/playlists.js
import axios from 'axios';

const API_URL = 'http://localhost:8000';

export const getPlaylists = async () => {
  const response = await axios.get(`${API_URL}/playlists`);
  return response.data;
};

export const getPlaylistDetails = async (playlistId) => {
  const response = await axios.get(`${API_URL}/playlists/${playlistId}`);
  return response.data;
};

export const createPlaylist = async (playlistData) => {
  const response = await axios.post(`${API_URL}/playlists`, playlistData);
  return response.data;
};

export const updatePlaylist = async (playlistId, playlistData) => {
  const response = await axios.put(`${API_URL}/playlists/${playlistId}`, playlistData);
  return response.data;
};

export const deletePlaylist = async (playlistId) => {
  await axios.delete(`${API_URL}/playlists/${playlistId}`);
};

export const addVideoToPlaylist = async (playlistId, videoId) => {
  const response = await axios.post(`${API_URL}/playlists/${playlistId}/videos`, { video_id: videoId });
  return response.data;
};

export const removeVideoFromPlaylist = async (playlistId, videoId) => {
  await axios.delete(`${API_URL}/playlists/${playlistId}/videos/${videoId}`);
};

File: playlists.py
import googleapiclient.errors
from typing import List

def get_playlists(youtube):
    try:
        request = youtube.playlists().list(
            part="snippet",
            mine=True,
            maxResults=25
        )
        response = request.execute()
        return response.get("items", [])
    except googleapiclient.errors.HttpError as error:
        print(f"An HTTP error occurred: {error}")
        return []

def create_playlist(youtube, title: str, description: str, tags: List[str]):
    try:
        request = youtube.playlists().insert(
            part="snippet,status",
            body={
                "snippet": {
                    "title": title,
                    "description": description,
                    "tags": tags,
                    "defaultLanguage": "en"
                },
                "status": {
                    "privacyStatus": "private"
                }
            }
        )
        response = request.execute()
        print(f"New playlist created: {response['id']}")
        return response
    except googleapiclient.errors.HttpError as error:
        print(f"An HTTP error occurred: {error}")
        return None

def update_playlist(youtube, playlist_id: str, title: str, description: str, tags: List[str]):
    try:
        request = youtube.playlists().update(
            part="snippet,status",
            body={
                "id": playlist_id,
                "snippet": {
                    "title": title,
                    "description": description,
                    "tags": tags,
                    "defaultLanguage": "en"
                },
                "status": {
                    "privacyStatus": "private"
                }
            }
        )
        response = request.execute()
        print(f"Playlist updated: {response['id']}")
        return response
    except googleapiclient.errors.HttpError as error:
        print(f"An HTTP error occurred: {error}")
        return None

def delete_playlist(youtube, playlist_id: str):
    try:
        request = youtube.playlists().delete(
            id=playlist_id
        )
        request.execute()
        print(f"Playlist deleted: {playlist_id}")
        return True
    except googleapiclient.errors.HttpError as error:
        print(f"An HTTP error occurred: {error}")
        return False